DatadogRUM/Sources/Feature/RUMDataStore.swift

```diff
@@ -0,0 +1,77 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+internal extension FeatureScope {
+    /// Data store endpoint suited for RUM data.
+    var rumDataStore: RUMDataStore {
+        RUMDataStore(featureScope: self)
+    }
+
+    /// RUM data store endpoint within SDK context.
+    func rumDataStoreContext(_ block: @escaping (DatadogContext, RUMDataStore) -> Void) {
+        dataStoreContext { context, dataStore in
+            block(context, rumDataStore)
+        }
+    }
+}
+
+/// RUM interface for data store.
+///
+/// It stores values in JSON format and implements convenience for type-safe key referencing and data serialization.
+/// Serialization errors are logged to telemetry.
+internal struct RUMDataStore {
```

Comment:
💡 This adds JSON-coding convenience for interacting with Data Store. It leverages SRP to separate concerns by abstracting error handling and failure reporting. It will be vastly leveraged for watchdog terminations, where the number of stored values will grow.

Reply:
I like that.
I did manual encoding/decoding in my PR using DataStore.

IMO worth adding similar kind of codable support extension as public. It's going to be very common use case.

cc @maxep

Reply:
👍 Sounds good. Let's move on with `scope.rumDataStore` in `RUM`. I can follow with refactoring PR to introduce `scope.jsonDataStore` in `DatadogInternal`.